File: platform/src/java/com/xcase/rest/generator/IProxyGenerator.java
package com.xcase.rest.generator;

import com.xcase.rest.generator.IAPIProxySettingsEndpoint;
import java.lang.invoke.MethodHandles;
import java.util.*;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public interface IProxyGenerator
{
    RESTServiceDefinition generateSourceString(IAPIProxySettingsEndpoint[] endpoints) throws Exception;
    RESTServiceDefinition generateSourceString(IAPIProxySettingsEndpoint endpoint, String document, String username, String password, String tenant) throws Exception;
    RESTServiceDefinition generateSourceString(IAPIProxySettingsEndpoint endpoint, String document) throws Exception;
    RESTServiceDefinition generateSourceString(String document) throws Exception;
    List<RESTServiceDefinition> generateSourceStringForRestServiceDefinitionList(IAPIProxySettingsEndpoint swaggerApiProxySettingsEndPoint, String swaggerDocument, String username, String password, String tenant) throws Exception;
}

File: dotnetcore/XCaseServiceClient/XCase.ProxyGenerator/IProxyGenerator.cs
namespace XCase.ProxyGenerator
{
    public interface IProxyGenerator
    {
        IServiceDefinition GenerateSourceString(IAPIProxySettingsEndpoint[] endpoints);
        IServiceDefinition GenerateSourceString(IAPIProxySettingsEndpoint endpoint, string document, string username, string password, string tenant);
        IServiceDefinition GenerateSourceString(IAPIProxySettingsEndpoint endpoint, string document);
        IServiceDefinition GenerateSourceString(string document);
    }
}
